client/src/components/ContentNavigator/utils.ts

```diff
@@ -106,3 +108,64 @@ export const isItemInRecycleBin = (item: ContentItem): boolean =>
   !!item && item.flags?.isInRecycleBin;
 
 export const isContentItem = (item): item is ContentItem => isValidItem(item);
+
+/**
+ * This attempts to create a new variabe in our SAS program using the following rules:
+ * - First, we attempt to use only the alphanumeric characters from contentItem.name as
+ *   as our variable name (without the file extension).
+ * - If that's empty, we instead use `<extension>file`
+ * - Next, we add a numeric suffix to the variable name, and check documentContent to make
+ *   sure what we're creating isn't already in use.
+ * - If the current variable is in use, we continue to increment our numeric suffix until
+ *   we find one that doesn't exist in our document.
+ *
+ * @param contentItemName ContentItem.name used for creating variable name.
+ * @param documentContent The contents of the editor we're dropping into.
+ *
+ * @returns string our new variable name.
+ */
+const extractVariableName = (
+  contentItemName: string,
+  documentContent: string,
+): string => {
+  const filePieces = parse(contentItemName);
+  const partialFileName = (
+    filePieces.name.replace(/[^a-zA-Z0-9]/g, "") ||
```

Comment:
underscore is also valid, isn't it?

Reply:
Yep! Updated in https://github.com/sassoftware/vscode-sas-extension/pull/510/commits/1e445783cf30d8ac6246174bf0b17a6bc8f5b209